test/fail_compilation/fail15089.d

```diff
@@ -0,0 +1,10 @@
+/*
+TEST_OUTPUT:
+---
+fail_compilation/fail15089.d(10): Error: cannot implicitly convert expression (130) of type int to byte
```

Comment:
If the start and end of the expression is stored the compiler can just read the original buffer.